### DISCUSSION TITLE
How would electron apps work?

### DISCUSSION THREAD

**Discussion by Milo123459** [community/community#3639] [Codespaces]
If I had an electron app I wanted to develop (ie, github desktop) and I ran it in codespaces, how would it work? Would it be able to run the electron app on my local machine?

**Comment by PranamBhat** [comment]
The way it works is — Electron takes a main file defined in your package. json file and executes it. This main file (usually named main. js) then creates application windows which contain rendered web pages with the added power of interacting with the native GUI (graphical user interface) of your operating system.

Check more here : [https://medium.com/developers-writing/building-a-desktop-application-with-electron-204203eeb658](https://medium.com/developers-writing/building-a-desktop-application-with-electron-204203eeb658)

**Reply by Milo123459** [reply]
Oh, but how would it work on Codespaces? I'm still confused

**Comment by matthewruzzi** [comment]
https://github.com/orgs/community/discussions/17420
